modules/mtl/arrow-mtl-data/src/test/kotlin/arrow/mtl/WriterTTest.kt

```diff
@@ -108,12 +114,18 @@ class WriterTTest : UnitSpec() {
         WriterT.monad(Option.monad(), ListK.monoid<Int>()),
         WriterT.genK(Option.genK(), Gen.list(Gen.int()).map { it.k() }),
         optionEQK()
+      ),
+
+      MonadStateLaws.laws(
+        WriterT.monadState<StateTPartialOf<ForId, Int>, String, Int>(String.monoid(), StateT.monadState(Id.monad())),
+        WriterT.genK(StateT.genK(Id.genK(), Gen.int()), Gen.string()),
+        WriterT.eqK(StateT.eqK(Id.eqK(), Int.eq(), Id.monad(), 0), String.eq())
       )
     )
   }
 }
 
-private fun <F, W> WriterT.Companion.genK(
+internal fun <F, W> WriterT.Companion.genK(
```

Comment:
These should publicly live in an `arrow-test`(-mtl) module?

Reply:
I think we could get away with putting them in `arrow-test` for now. That project depends on mtl-data, recursion-data, fx etc already anyway.

I think at some point that should be split apart, but that could be quite a big change as lots of tests could transiently depend on that behaviour...

Reply:
Moved them to `arrow-test` for now